### ISSUE TITLE
Next 9 - File to import not found or unreadable: _variables.sass.

### ISSUE THREAD

**Issue by RichAWarren** [couds/react-bulma-components#232]
Hi! I've followed the advanced instructions to install the components for next.js but am still failing to correctly import the variable sass file. Its a next.js 9 app created with creat-next-app. On `yarn dev` I get the following stack trace:

```
ModuleBuildError: Module build failed (from ./node_modules/@zeit/next-css/node_modules/mini-css-extract-plugin/dist/loader.js):
ModuleBuildError: Module build failed (from ./node_modules/sass-loader/lib/loader.js):

@import "_variables.sass"
^
      File to import not found or unreadable: _variables.sass.
      in /Users/richard/Documents/code/oddsjar/node_modules/react-bulma-components/src/components/utils.sass (line 3, column 1)
    at /Users/richard/Documents/code/oddsjar/node_modules/webpack/lib/NormalModule.js:316:20
    at /Users/richard/Documents/code/oddsjar/node_modules/loader-runner/lib/LoaderRunner.js:367:11
    at /Users/richard/Documents/code/oddsjar/node_modules/loader-runner/lib/LoaderRunner.js:233:18
    at context.callback (/Users/richard/Documents/code/oddsjar/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
    at Object.callback (/Users/richard/Documents/code/oddsjar/node_modules/sass-loader/lib/loader.js:55:13)
    at Object.<anonymous> (/Users/richard/Documents/code/oddsjar/node_modules/async/dist/async.js:2271:31)
    at Object.callback (/Users/richard/Documents/code/oddsjar/node_modules/async/dist/async.js:969:16)
    at options.error (/Users/richard/Documents/code/oddsjar/node_modules/node-sass/lib/index.js:294:32)
    at /Users/richard/Documents/code/oddsjar/node_modules/webpack/lib/NormalModule.js:316:20
    at /Users/richard/Documents/code/oddsjar/node_modules/loader-runner/lib/LoaderRunner.js:367:11
    at /Users/richard/Documents/code/oddsjar/node_modules/loader-runner/lib/LoaderRunner.js:182:20
    at context.callback (/Users/richard/Documents/code/oddsjar/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
    at /Users/richard/Documents/code/oddsjar/node_modules/@zeit/next-css/node_modules/mini-css-extract-plugin/dist/loader.js:112:14
    at /Users/richard/Documents/code/oddsjar/node_modules/webpack/lib/Compiler.js:343:11
    at /Users/richard/Documents/code/oddsjar/node_modules/webpack/lib/Compiler.js:681:15
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/richard/Documents/code/oddsjar/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:24:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/richard/Documents/code/oddsjar/node_modules/tapable/lib/Hook.js:154:20)
    at /Users/richard/Documents/code/oddsjar/node_modules/webpack/lib/Compiler.js:678:31
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/richard/Documents/code/oddsjar/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/richard/Documents/code/oddsjar/node_modules/tapable/lib/Hook.js:154:20)
    at /Users/richard/Documents/code/oddsjar/node_modules/webpack/lib/Compilation.js:1423:35
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/richard/Documents/code/oddsjar/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/richard/Documents/code/oddsjar/node_modules/tapable/lib/Hook.js:154:20)
    at /Users/richard/Documents/code/oddsjar/node_modules/webpack/lib/Compilation.js:1414:32
```

I've tried to follow all the similar issues in the repo but none have yet solved the problem, the only difference I can see is that I am using nextjs 9. I've setup a demo repo here https://github.com/RichAWarren/react-bulma-next-9

Thanks in advance!

**Comment by RichAWarren** [comment]
Update to this:
I suspect the problem is down to nextjs 9.2 significantly changing the way css is imported, see details:
https://nextjs.org/blog/next-9-2. Will experiment with fixes in due course

**Comment by mikecanann** [comment]
I'm running into the same issue. But I'm not using next.

**Comment by MrCreeper1008** [comment]
@mikecanann what framework are you using?

**Comment by MrCreeper1008** [comment]
@RichAWarren I have no experience in next.js, but I'll look into this issue in my spare time. Thank you!

**Comment by mikecanann** [comment]
I'm just using react. I created a project using create-react-app

**Comment by MrCreeper1008** [comment]
Did you follow this [instruction](https://github.com/couds/react-bulma-components#cra)?

**Comment by philohelp** [comment]
Yes I have the same issue with NextJS current version, cannot find the mean to make it work with bulma components.

**Comment by couds** [comment]
I't not ideal but something you can do is import from the components without the sass 
```
import { Button } from 'react-bulma-components';
...
```
And import the styles directly from `bulma`.

**Comment by mrd9483** [comment]
After following all of the documentation, changing `@import "_variables.sass"` to `@import "~_variables.sass"` in src/components/utils.sass works.

It seems to have something to do with the aliasing in next.js. 

The solution above isn't good, as I'm directly changing a file within the node module, and I don't know enough about this to make an informed pull request.

**Comment by MrCreeper1008** [comment]
thanks for reporting this. i'll look into this and see what i can do.

**Comment by couds** [comment]
Hi.

I was taking a look and I manage to fix this doing this:

1. create the `_variables.sass` in the `pages` folder
2. Modify or create next.config.js with

```js
const withSass = require('@zeit/next-sass')
const withTM = require('next-plugin-transpile-modules');

module.exports = withTM(withSass({
    transpileModules: ['react-bulma-components'], // Tell next to transpile the react-bulma-components module
    // NOTE: this also allow to use abolute import from this folder not only for the _variables.sass file
    webpack: (config, { buildId, dev, isServer, defaultLoaders, webpack }) => {
        config.resolve.modules.push('./pages'); // Add to webpack configuration the folder where you put the _variables file
        return config
    },
}))
```

**Comment by mrd9483** [comment]
Hey @couds thanks for your work!

After following the new instructions, I'm still getting the error. here's a link to a repo I created, with a fresh next.js, react-bulma-components, and directions from the README.md:

https://github.com/mrd9483/react-bulma-component-test

This is a freshly created next.js install with bulma plugin. Once everything is set up, the only way I can get this to work is to go into node_modules and change line 3 of util.scss under react-bulma-components > src > components to: `@import "~_variables.sass"`

Without it, I get:

./node_modules/react-bulma-components/src/components/button/button.sass @import "_variables.sass" ^ File to import not found or unreadable: _variables.sass. in /Users/matthewdoyle/Dev/react-bulma-component-test/node_modules/react-bulma-components/src/components/utils.sass (line 3, column 1)

Because I don't know the impact on frameworks other than next, not sure if that's a reasonable answer.

**Comment by couds** [comment]
Hi @mrd9483 

I'm checking out the repo and I don't know why is not working. I created another repo and follow the instructions and works but in your repo I can't get the configuration to work. 

[Here](https://github.com/couds/nextjs-react-bulma-components) I created a basic example of the configuration working I followed this [instructions](https://nextjs.org/learn/basics/create-nextjs-app/setup). Can you try again creating a project this way? 

I will take a look on your project to see what's the difference and why its not working.

Sorry for the inconvenience

**Comment by mrd9483** [comment]
Hi @couds 

Thanks for the repo you sent... I still could not get it working unfortunately, same bug :(

I tried on Mac OS X, Ubuntu and Windows 10, and seeing the same thing on each. Changing the _variables.sass in util.sass to ~_variables.sass makes it work.

**Comment by mrd9483** [comment]
I had a few moments to play around, and I figured it out. The documentation references next-transpile-modules but the code references next-plugin-transpile-modules. I have a pull request in to correct the documentation.

**Comment by humet** [comment]
Hi, After reading through the above and following the instructions I'm still having this issue. I created an example repo here:

https://github.com/humet/drive-in 

Could it be something to do with Next.JS 9.4.2? 

On another note, will this package work with the new Next.js in-built SASS compiler? I tried, but I guess the only way is to rename all the SASS to follow the .module.sass format it requires.

**Comment by mrd9483** [comment]
@humet 

I spent around an hour playing with your code, and there seems to be an inconsistency with the way _variables.sass is served up, at least in next. I tried erasing node_modules and .next folders and rebuilding/running a bunch, and it would work 10% of the time.

What I was able to eventually get to work consistently was move _variables.sass to the root folder. I was able to build and run every time this way. I tried it on your project and mine as well, seems to be doing the trick. Would love to know why the inconsistent behavior though...

**Comment by troyshu** [comment]
I ran into this issue as well, also trying to use the React bulma components here with nextjs. @mrd9483`'s latest comment worked for me (`next build` no longer errors out with `File to import not found or unreadable: _variables.sass`. Weirdly, `next dev` never gave the same error):

1. Move `_variables.sass` to the root folder
2. Remove the line `config.resolve.modules.push('./pages');` in your `next.config.js` file. I don't know why this works, but leaving this line in and changing `'./pages'` to `./` caused the bulma styles to completely ignore the variables I had set/overridden in `_variables.sass`... because the root directory doesn't need to be module? Either way, my sass variable overrides worked again after removing this line.
